tests_e2e/orchestrator/scripts/install-agent

```diff
@@ -76,9 +75,14 @@ echo "Service name: $service_name"
 #
 # Install the package
 #
-echo "Installing $package..."
-unzip -d "/var/lib/waagent/WALinuxAgent-$version" -o "$package"
-sed -i 's/AutoUpdate.Enabled=n/AutoUpdate.Enabled=y/g' /etc/waagent.conf
+echo "Installing $package as version $version..."
+unzip.py "$package" "/var/lib/waagent/WALinuxAgent-$version"
+
+# Ensure that AutoUpdate is enabled. some distros, e.g. Flatcar, don't have a waagent.conf
+# but AutoUpdate defaults to True so there is no need to anything in that case.
+if [[ -e /etc/waagent.conf ]]; then
```

Comment:
The default values are hardcoded in the agent.Specific tests may need to create /etc/waagent.conf, but the overall infrastructure should not create it, since that is not the way actual VMs would look like.

BTW, currently VMAccess breaks on Flatcar because it has a hard dependency on /etc/waagent.conf; the extension needs to fix this dependency, since user-created VMs won't have it. If the AgentTestSuite created the conf file, we would be masking this bug.

Reply:
But why flatcar vm don't have it? what is special about it? My suspicion is that our agent source setup itself don't have it. In general distro owners copy or look at our agent setup to bootstrap agent right.

Reply:
That is the way they setup Flatcar and is perfectly valid. Flatcar is supposed to be immutable so I imagine they are not allowing for any configuration changes by default,